doc/rpc/pathfinder_ws.json

```diff
@@ -10,19 +10,55 @@
             "name": "pathfinder_subscribe",
             "summary": "Open a new websocket subscription",
             "description": "Creates a websocket stream which will fire events of the requested subscription type. Only available for websocket connections.",
-            "params": [
-                {
-                    "name": "kind",
-                    "summary": "The type of subscription",
-                    "required": true,
-                    "schema": {
-                        "type": "string",
-                        "enum": [
-                            "newHeads"
-                        ]
-                    }
-                }
-            ],
+            "params": {
+                "oneOf": [
+                    [
+                        {
+                            "name": "kind",
+                            "summary": "The type of subscription",
+                            "required": true,
+                            "schema": {
+                                "type": "string",
+                                "enum": [
+                                    "newHeads"
+                                ]
+                            }
+                        }
+                    ],
+                    [
+                        {
+                            "name": "kind",
+                            "summary": "The type of subscription",
+                            "required": true,
+                            "schema": {
+                                "type": "string",
+                                "enum": [
+                                    "events"
+                                ]
+                            }
+                        },
+                        {
+                            "name": "address",
+                            "summary": "The from_address to filter events by",
+                            "required": false,
+                            "schema": {
+                                "$ref": "#/components/schemas/FELT"
+                            }
+                        },
+                        {
+                            "name": "keys",
+                            "summary": "The keys to filter events by",
+                            "required": false,
+                            "schema": {
+                                "type": "array",
+                                "items": {
+                                    "$ref": "#/components/schemas/FELT"
+                                }
+                            }
+                        }
+                    ]
+                ]
+            },
```

Comment:
This approach is type safer, the other option is to compress the spec a bit instead - making it potentially more readible?

Maybe something like
```json
{
  "name": "kind",
  "summary": "The type of subscription",
  "required": true,
  "schema": {
    "type": "string",
    "enum": [
      "newHeads",
      "events"
    ]
},
{
  "name": "events_filter",
  "summary": "Required for `events` subscriptions",
  "required": false,
  ...
}
```
with a separate optional configuration parameter for the event filter? This would add a new optional configuration parameter for every new subscription (in some cases they're null).

Or alternatively, a new "config" field which is an enum over the all configuration options.

I'm not entirely certain which I favor tbh. Maybe type safer is better. I just dislike the spec bloat that happens. But maybe that is just what it is.